src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Registry/PluginRegistry.cs

```diff
@@ -46,8 +46,23 @@ public PluginRegistry(string registryRoot)
             this.registryFilePath = Path.Combine(registryRoot, registryFileName);
             this.backupRegistryFilePath = this.registryFilePath + backupRegistryFileExtension;
             this.lockFilePath = Path.Combine(registryRoot, lockFileName);
+            this.FileLock = new FileDistributedLock(new FileInfo(this.lockFilePath));
         }
 
+        /// <summary>
+        ///    An exclusive file lock that indicates whether a process is currently interacting with the registry.
+        ///    Always make sure to acquire this lock before interacting with the registry and release it after the interaction.
+        ///    Usage:
+        ///    <code>
+        ///         await using (await myLock.AcquireAsync(...))
+        ///         {
+        ///             /* we have the lock! */
+        ///         }
+        ///         // dispose releases the lock
```

Comment:
nit : mind showing how to call dispose instead of a code comment? e.g. myLock.dispose()